MODEL_SCALABILITY.md

```diff
@@ -0,0 +1,71 @@
+# Document for model scalability
+- author: [Kevin Ko](https://github.com/hyunwoongko)
+
+## 1. Activation Checkpointing
+I added activation checkpointing to save activation memory. 
+You can use this feature by specifying `gradient_checkpointing` in the `train_job`.
+
+Example:
+
+```
+gradient_checkpointing: true
+```
+
+## 2. DeepSpeed Training
+
+Run the following instruction.
+
+```
+python -m torch.distributed.launch \
+       --nproc_per_node NUM_YOUR_GPUS \
+       -m carp.pytorch.training.train \
+       --data_path="carp/dataset" \
+       --config_path ./configs/YOUR_CONFIG_NAME.yml \
+       --deepspeed_config ./configs/scalability/YOUR_DS_CONFIG_NAME.json
+```
+
+## 3. Integrated architectures
+- `carp.py`
+- `carp_cloob.py`
+- `carp_coop.py`
+- `carp_shared_encoder.py`
+
+`carp_mlm.py` and `carp_momentum.py` are not integrated yet.
```

Comment:
We don't need to mention these, these are both failed experiments.

Reply:
I see

Reply:
Then how about remove these. or move to legacy folder?

Reply:
Going to do this in a future PR